drivers/photo-electronic/adpd410x/adpd410x.h

```diff
@@ -0,0 +1,1244 @@
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*******************************************************************************/
+
+#ifndef ADPD410X_H_
+#define ADPD410X_H_
+
+/******************************************************************************/
+/***************************** Include Files **********************************/
+/******************************************************************************/
+
+#include <stdbool.h>
+#include "spi.h"
+#include "i2c.h"
+#include "gpio.h"
+
+/******************************************************************************/
+/********************** Macros and Constants Definitions **********************/
+/******************************************************************************/
+
+#define ADPD410X_REG_FIFO_STATUS	0x0000
+#define ADPD410X_REG_INT_STATUS_DATA	0x0001
+#define ADPD410X_REG_INT_STATUS_LEV0	0x0002
+#define ADPD410X_REG_INT_STATUS_LEV1	0x0003
+#define ADPD410X_REG_FIFO_TH		0x0006
+#define ADPD410X_REG_INT_ACLEAR		0x0007
+#define ADPD410X_REG_CHIP_ID		0x0008
+#define ADPD410X_REG_OSC32M		0x0009
+#define ADPD410X_REG_OSC32M_CAL		0x000A
+#define ADPD410X_REG_OSC1M		0x000B
+#define ADPD410X_REG_OSC32K		0x000C
+#define ADPD410X_REG_TS_FREQ		0x000D
+#define ADPD410X_REG_TS_FREQH		0x000E
+#define ADPD410X_REG_SYS_CTL		0x000F
+#define ADPD410X_REG_OPMODE          	0x0010
+#define ADPD410X_REG_STAMP_L         	0x0011
+#define ADPD410X_REG_STAMP_H         	0x0012
+#define ADPD410X_REG_STAMPDELTA      	0x0013
+#define ADPD410X_REG_INT_ENABLE_XD   	0x0014
+#define ADPD410X_REG_INT_ENABLE_YD   	0x0015
+#define ADPD410X_REG_INT_ENABLE_XL0  	0x0016
+#define ADPD410X_REG_INT_ENABLE_XL1	0x0017
+#define ADPD410X_REG_INT_ENABLE_YL0	0x001a
+#define ADPD410X_REG_INT_ENABLE_YL1	0x001b
+#define ADPD410X_REG_FIFO_STATUS_BYTES	0x001e
+#define ADPD410X_REG_INPUT_SLEEP     	0x0020
+#define ADPD410X_REG_INPUT_CFG       	0x0021
+#define ADPD410X_REG_GPIO_CFG        	0x0022
+#define ADPD410X_REG_GPIO01          	0x0023
+#define ADPD410X_REG_GPIO23          	0x0024
+#define ADPD410X_REG_GPIO_IN         	0x0025
+#define ADPD410X_REG_GPIO_EXT           0x0026
+#define ADPD410X_REG_DATA_HOLD_FLAG  	0x002E
+#define ADPD410X_REG_FIFO_DATA       	0x002F
+#define ADPD410X_REG_SIGNAL1_L(ts)     	(0x0030 + ts * 0x08)
+#define ADPD410X_REG_SIGNAL1_H(ts)     	(0x0031 + ts * 0x08)
+#define ADPD410X_REG_SIGNAL2_L(ts)     	(0x0032 + ts * 0x08)
+#define ADPD410X_REG_SIGNAL2_H(ts)     	(0x0033 + ts * 0x08)
+#define ADPD410X_REG_DARK1_L(ts)       	(0x0034 + ts * 0x08)
+#define ADPD410X_REG_DARK1_H(ts)       	(0x0035 + ts * 0x08)
+#define ADPD410X_REG_DARK2_L(ts)       	(0x0036 + ts * 0x08)
+#define ADPD410X_REG_DARK2_H(ts)       	(0x0037 + ts * 0x08)
+#define ADPD410X_REG_IO_ADJUST        	0x00B4
+#define ADPD410X_REG_I2C_KEY         	0x00B6
+#define ADPD410X_REG_I2C_ADDR        	0x00B7
+#define ADPD410X_REG_TS_CTRL(ts)       	(0x0100 + ts * 0x20)
```

Comment:
Always paranthesize macro parameters in C, this is a potential bug.
What if uses passes `ts` as arithmetic ? This gets expanded to `0x0100 + a + b * 0x20` which is not what you want. If you use parantheses `0x0100 + (a + b) * 0x20` would give correct result.